src/lib/comms/sol-oic-server.c

```diff
@@ -722,9 +726,21 @@ _sol_oic_resource_type_handle(
         return -1;
     }
 
-    if (!handle_fn) {
-        code = SOL_COAP_RSPCODE_NOT_IMPLEMENTED;
+    len = strlen(response_prefix);
+
+    if (sol_coap_packet_get_payload(response, &response_payload, &response_len) < 0)
         goto done;
+
+    if (response_len <= (len + strlen(response_suffix)))
+        goto done;
+
+    memcpy(response_payload, response_prefix, len);
+    response_payload += len;
+    response_len -= len;
```

Comment:
you shouldn't need this. Use `sol_buffer` functions, they will append strings for you. If you set the buffer to not-owned/fixed capacity, then no resize is done and you can just check for errors if they happen.